src/deploy/osp_deployer/settings/config.py

```diff
@@ -1,528 +1,561 @@
+            '/pilot/deploy-overcloud.py'
+        self.assign_role_py = self.foreman_configuration_scripts +\
+            '/pilot/assign_role.py'
+        self.network_env_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/network-environment.yaml'
+        self.dell_storage_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/dell-cinder-backends.yaml'
+        self.dell_env_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/dell-environment.yaml'
+        self.neutron_ovs_dpdk_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/neutron-ovs-dpdk.yaml'
+        self.static_ips_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/static-ip-environment.yaml'
+        self.static_vip_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/static-vip-environment.yaml'
+        self.sanity_ini = self.foreman_configuration_scripts + \
+            '/pilot/deployment-validation/sanity.ini'
+        self.ipxe_rpm = self.foreman_configuration_scripts + \
+            '/pilot/ipxe/ipxe-bootimgs-20151005-1.git6847232.el7.' \
+            'test.noarch.rpm'
+
+        # The NIC configurations settings are validated after the Settings
+        # class has been instanciated.  Guard against the case where the two
+        # fixed are missing here to prevent an exception before validation
+        nics_settings = self.get_nics_settings()
+        if 'nic_env_file' in nics_settings:
+            self.nic_env_file = nics_settings['nic_env_file']
+            self.nic_env_file_path = self.foreman_configuration_scripts + \
+                '/pilot/templates/nic-configs/' + self.nic_env_file
+        if 'sah_bond_opts' in nics_settings:
+            self.sah_bond_opts = nics_settings['sah_bond_opts']
+
+        # This particular section has been moved right after the nics_settings
+        # section in order to catch the mode used by the nic-environment file
+        dellnfv_settings = self.get_settings_section(
+            "Dell NFV Settings")
+        self.ovs_dpdk_enable = dellnfv_settings[
+            'ovs_dpdk_enable']
+        self.ovs_dpdk_role_list = dellnfv_settings[
+            'ovs_dpdk_role_list']
+        self.ovs_dpdk_policy = dellnfv_settings[
+            'ovs_dpdk_policy']
+        self.enable_ovs_dpdk = False
+        if self.ovs_dpdk_enable.lower() == 'false':
+            pass
+        elif self.ovs_dpdk_enable.lower() == 'true':
+            self.enable_ovs_dpdk = True
+            for each in re.split(r'[_/]', self.nic_env_file):
+                if each.find('mode') != -1:
+                    self.ovs_dpdk_mode = each[-1:]
+        else:
+            raise AssertionError('Only supported values for '
+                                 'ovs_dpdk_enable are true or false. ')
+        if self.enable_ovs_dpdk:
+            mode = str(self.ovs_dpdk_mode)
```

Comment:
If you make the above change, then mode will be a string already, so the above line won't be needed.

Reply:
The ovs_dpdk_mode was already a string, this line is therefore useless.